src/ripple/app/tx/impl/CancelTicket.cpp

```diff
@@ -41,49 +42,23 @@ class CancelTicket
 
     }
 
-    TER doApply () override
+    TER preCheck () override
     {
-        assert (mTxnAccount);
-
-        uint256 const ticketId = mTxn.getFieldH256 (sfTicketID);
-
-        SLE::pointer sleTicket = mEngine->view ().entryCache (ltTICKET, ticketId);
-
-        if (!sleTicket)
-            return tecNO_ENTRY;
-
-        auto const ticket_owner =
-            sleTicket->getFieldAccount160 (sfAccount);
-
-        bool authorized =
-            mTxnAccountID == ticket_owner;
-
-        // The target can also always remove a ticket
-        if (!authorized && sleTicket->isFieldPresent (sfTarget))
-            authorized = (mTxnAccountID == sleTicket->getFieldAccount160 (sfTarget));
+        // A Ticket may not be used to cancel a Ticket.  So a Sequence of
+        // zero (which indicates that a Ticket is being used) is malformed.
+        if (mTxn.getSequence () == 0)
+            return tefPAST_SEQ;
```

Comment:
Yeah, my preference would be to return a `temBAD_SEQUENCE` here, as Nik noted.  However, historically, the zero value was not explicitly checked.  So the old behavior for any transaction seeing a Sequence of zero would be to return `tefPAST_SEQ`.  Vinnie asked me to adjust the pull request so it was not transaction changing.  Returning a `tefPAST_SEQ` here (and a couple of other places) was part of that effort.

You folks can get a sense of the nature of the change by looking at this commit: https://github.com/ripple/rippled/commit/4a898a91f881ea920b95d619a3d35d489118bcb1

Personally, I prefer returning `temBAD_SEQUENCE`.  A zero Sequence without a Ticket can never succeed.  But it is, literally, transaction changing.  But it's only transaction changing if someone messes up big time by giving a transaction a zero sequence and no Ticket.  You guys can decide on the right call to make.